### PR TITLE
[SPARK-7918] [MLlib] MLlib Python doc parity check for evaluation and feature

### PR DESCRIPTION
**Pull request by yanboliang**
Check then make the MLlib Python evaluation and feature doc to be as complete as the Scala doc.

### CODE REVIEW

File: python/pyspark/mllib/evaluation.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ class BinaryClassificationMetrics(JavaModelWrapper):
     """
     Evaluator for binary classification.
 
+    :param scoreAndLabels: an RDD of (score, label) pairs
```
Comment (mengxr) [inline python/pyspark/mllib/evaluation.py]: Great! I didn't know this is the right way to make them show up in the doc. Is there an official doc for this?

File: python/pyspark/mllib/evaluation.py [hunk 2]
```diff
@@ -76,6 +75,8 @@ class RegressionMetrics(JavaModelWrapper):
     """
     Evaluator for regression.
 
+    :param predictionAndObservations: an RDD of (prediction, observation) pairs.
```
Comment (mengxr) [inline python/pyspark/mllib/evaluation.py]: This line is too wide for docstring. PEP8 standard is 72.

File: python/pyspark/mllib/evaluation.py [hunk 3]
```diff
@@ -277,6 +274,8 @@ class RankingMetrics(JavaModelWrapper):
     """
     Evaluator for ranking algorithms.
 
+    :param predictionAndLabels: an RDD of (predicted ranking, ground truth set) pairs.
```
Comment (mengxr) [inline python/pyspark/mllib/evaluation.py]: line too wide

File: python/pyspark/mllib/evaluation.py [hunk 4]
```diff
@@ -347,6 +343,9 @@ class MultilabelMetrics(JavaModelWrapper):
     """
     Evaluator for multilabel classification.
 
+    :param predictionAndLabels: an RDD of (predictions, labels) pairs,
+                                both are non-null Arrays, each with unique elements.
```
Comment (mengxr) [inline python/pyspark/mllib/evaluation.py]: perhaps this one is too wide as well

File: python/pyspark/mllib/feature.py [hunk 5]
```diff
@@ -92,9 +91,9 @@ def transform(self, vector):
         """
         Applies unit length normalization on a vector.
 
-        :param vector: vector or RDD of vector to be normalized.
+        :param   vector: vector or RDD of vector to be normalized.
```
Comment (mengxr) [inline python/pyspark/mllib/feature.py]: Is the extra sparse necessary?

File: python/pyspark/mllib/feature.py [hunk 6]
```diff
@@ -164,6 +163,13 @@ class StandardScaler(object):
     variance using column summary statistics on the samples in the
     training set.
 
+    :param withMean: False by default. Centers the data with mean
+                     before scaling. It will build a dense output, so this
+                     does not work on sparse input and will raise an
+                     exception.
+    :param withStd:  True by default. Scales the data to unit
```
Comment (mengxr) [inline python/pyspark/mllib/feature.py]: We don't do vertical alignment in Spark code. So we should remove one space after `withStd:`.

File: python/pyspark/mllib/evaluation.py (hunk 1 above)
Comment (yanboliang) [inline python/pyspark/mllib/evaluation.py]: You can refer http://stackoverflow.com/questions/5599254/how-to-use-sphinxs-autodoc-to-document-a-classs-init-self-method . I'm also considering which is a better and unified way for spark.

**Comment by jkbradley** [comment]
LGTM, I'll merge this with master and branch-1.4
Thanks!

**Pull request closed without merging**